File: rules/csharp/tests/cs.quality.empty-catch/Catch.cs
class CatchTests {
  void Swallow() {
    try { Risky(); } catch (Exception e) {} // crit:expect cs.quality.empty-catch
  }
  void SwallowBare() {
    try { Risky(); } catch {} // crit:expect cs.quality.empty-catch
  }
  void Handled() {
    try { Risky(); } catch (Exception e) { Log(e); } // crit:expect-not cs.quality.empty-catch
  }
  void Rethrows() {
    try { Risky(); } catch (Exception e) { throw; } // crit:expect-not cs.quality.empty-catch
  }
}

File: rules/javascript/tests/js.quality.empty-catch/catch.js
function swallow() {
  try { risky(); } catch (e) {} // crit:expect js.quality.empty-catch
}

function swallowNoBinding() {
  try { risky(); } catch {} // crit:expect js.quality.empty-catch
}

function handled() {
  try { risky(); } catch (e) { report(e); } // crit:expect-not js.quality.empty-catch
}

function rethrows() {
  try { risky(); } catch (e) { throw e; } // crit:expect-not js.quality.empty-catch
}
